core/src/com/unciv/models/ruleset/unique/IHasUniques.kt

```diff
@@ -153,9 +153,28 @@ interface IHasUniques : INamed {
     ): Boolean {
         if (hasUnique(UniqueType.HiddenFromCivilopedia)) return true
         if (gameInfo != null && isUnavailableBySettings(gameInfo)) return true
-        if (gameInfo == null && ruleset!!.beliefs.isEmpty()) return true
+        if (gameInfo == null && ruleset != null) {
+            if (!gameAgnosticVisibleForReligionOrEspionage(
+                    ruleset.beliefs.isNotEmpty(),
+                    UniqueType.ConditionalReligionEnabled,
+                    UniqueType.ConditionalReligionDisabled
+                )) return true
+            if (!gameAgnosticVisibleForReligionOrEspionage(
+                    ruleset.nations.values.any { it.spyNames.isNotEmpty() },
+                    UniqueType.ConditionalEspionageEnabled,
+                    UniqueType.ConditionalEspionageDisabled
+                )) return true
+        }
         return false
     }
     /** Overload of [isHiddenFromCivilopedia] for use in actually game-agnostic parts of Civilopedia */
     fun isHiddenFromCivilopedia(ruleset: Ruleset) = isHiddenFromCivilopedia(UncivGame.getGameInfoOrNull(), ruleset)
+
+    private fun gameAgnosticVisibleForReligionOrEspionage(hasFeature: Boolean, enabler: UniqueType, disabler: UniqueType): Boolean {
```

Comment:
I have no idea what this is meant to do

Reply:
Coming back to this, I still can't read this.
What does this function mean? What is 'agnostic visible'? What is the bad case and what is the fix?

Reply:
So like... shouldBeVisibleIfNoGameLoaded?